server/channels/app/platform/metrics.go

```diff
@@ -247,3 +252,52 @@ func (ps *PlatformService) Metrics() einterfaces.MetricsInterface {
 
 	return ps.metricsIFace
 }
+
+func (ps *PlatformService) CreateCPUProfile() (*model.FileData, error) {
+	var b bytes.Buffer
+
+	err := rpprof.StartCPUProfile(&b)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to start CPU profile")
+	}
+
+	time.Sleep(cpuProfileDuration)
```

Comment:
question, will this block execution? if so, wouldn't it make sense to run this logic in a separate routine?

Reply:
Sleep blocks execution, which is the whole point. `StartCPUProfile` starts the CPU profile generation, and then we wait for as long as the profile should capture. A longer sleep means the generated CPU profile contains a longer timeframe.

Reply:
got it, thank you for the clarification.